### PR TITLE
EUSS HP-RTU measure updates

### PR DESCRIPTION
**Pull request by JanghyunJK**
Pull request overview
---------------------

@ChristopherCaradonna @carlobianchi89 this is the latest HPRTU measure PR!

This became a very complicated change including multiple sub-PRs and multiple updates on the same thing. But below are the summary of changes:

- `upgrade_hvac_add_heat_pump_rtu` measure (and corresponding unit tests) refactored and updated

- `comstock_sensitivity_reports` measure updated to include cycling impacts

- these sub-PRs include some major context of the changes
  - https://github.com/NREL/ComStock/pull/245
  - https://github.com/NREL/ComStock/pull/222
  - https://github.com/NREL/ComStock/pull/204 
  - https://github.com/NREL/ComStock/pull/198
  - https://github.com/NREL/ComStock/pull/167
  -  https://github.com/NREL/ComStock/pull/166
  -  https://github.com/NREL/ComStock/pull/177
  - https://github.com/NREL/ComStock/pull/162
  - also including Chris' updates regarding Cold Climate Heat Pump Challenge updates
  
- this PR also includes some other stuffs: `create_` measure update to 3.8, os stds update to 0.6.1, and maybe more... 

- updated some shared/test models to 3.8

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [x] Upgrade Measures
 - [x] Reporting Measures
 - [ ] Postprocessing

Author pull request checklist:

 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Added tests for new measures
 - [x] Updated measure .xml(s)
 - [x] Register values added to `comstock_column_definitions.csv`
 - [x] Both `options_lookup.tsv` files updated
 - [ ] 10k+ test run
 - [ ] Change documentation written
 - [x] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [ ] Added 'See ComStock License' language to first two lines of each code file
 - [ ] Implements corresponding measure tests and indexing path in `test/measure_tests.txt` or/and `test/resource_measure_tests.txt`
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

### CODE REVIEW

File: measures/comstock_sensitivity_reports/measure.rb [hunk 1]
```diff
@@ -2746,9 +2746,9 @@ def run(runner, user_arguments)
         coil_heating = airloopunisys.heatingCoil.get
         # get appropriate cycling ratio timeseries depending on coil type
         ts_cycling_ratio = nil
-        if coil_cooling.to_CoilCoolingDXTwoSpeed.is_initialized
+        if coil_heating.to_CoilCoolingDXTwoSpeed.is_initialized
```
Comment (ChristopherCaradonna) [inline measures/comstock_sensitivity_reports/measure.rb]: @JanghyunJK to_CoilCoolingDXTwoSpeed -> should these be to_Coil**_Heating_**DXTwoSpeed?

File: measures/comstock_sensitivity_reports/measure.xml [hunk 2]
```diff
@@ -1,6 +1,13 @@
 <?xml version="1.0"?>
 <measure>
   <schema_version>3.1</schema_version>
+  <error>C:/Users/ccaradon/Documents/GitHub/ComStock/measures/comstock_sensitivity_reports/measure.rb:2244: syntax error, unexpected constant, expecting `end'
```
Comment (ChristopherCaradonna) [inline measures/comstock_sensitivity_reports/measure.xml]: @JanghyunJK - does the XML need to be updated?

File: measures/comstock_sensitivity_reports/README.md
Comment (ChristopherCaradonna) [inline measures/comstock_sensitivity_reports/README.md:1]: Do you need to regenerate the XML? Some of the attributes are being deleted.

File: postprocessing/comstockpostproc/comstock.py
Comment (ChristopherCaradonna) [inline postprocessing/comstockpostproc/comstock.py:1]: Should this be included?

File: resources/Gemfile
Comment (ChristopherCaradonna) [inline resources/Gemfile:1]: Should this be included?

File: test/reporting_measure_tests.txt
Comment (ChristopherCaradonna) [inline test/reporting_measure_tests.txt:1]: Should this be included?

File: test/upgrade_measure_tests.txt
Comment (ChristopherCaradonna) [inline test/upgrade_measure_tests.txt:1]: Should this be included?

File: test/workflow_measure_tests.txt
Comment (ChristopherCaradonna) [inline test/workflow_measure_tests.txt:1]: Should this be included?

File: Rakefile
Comment (ChristopherCaradonna) [inline Rakefile:1]: Should these changes be included?

File: README.md
Comment (ChristopherCaradonna) [inline README.md:1]: Should these changes be included?

File: .rubocop.yml
Comment (ChristopherCaradonna) [inline .rubocop.yml:1]: Should this be included?

File: .gitignore
Comment (ChristopherCaradonna) [inline .gitignore:1]: Should this be included?

**Review by ChristopherCaradonna — COMMENTED** [review]
Overall looks good but there are some files that I am unsure if we should be changing. Just confirm that they are intentional before we merge.

File: resources/measures/upgrade_hvac_add_heat_pump_rtu/measure.rb [hunk 3]
```diff
@@ -336,94 +380,851 @@ def model_add_curve(model, curve_name, standards_data_curve, std)
...
+      'Site Outdoor Air Drybulb Temperature',
+      # 'Heating Coil Crankcase Heater Electricity Rate',
+      # 'Heating Coil Defrost Electricity Rate'
+    ]
+    out_vars.each do |out_var_name|
+        ov = OpenStudio::Model::OutputVariable.new('ov', model)
+        ov.setKeyValue('*')
+        ov.setReportingFrequency('hourly')
+        ov.setVariableName(out_var_name)
+    end
```
Comment (eringold) [inline resources/measures/upgrade_hvac_add_heat_pump_rtu/measure.rb]: comment out

File: resources/measures/upgrade_hvac_add_heat_pump_rtu/measure.rb [hunk 4]
```diff
@@ -336,94 +380,851 @@ def model_add_curve(model, curve_name, standards_data_curve, std)
...
+
+      # Return interpolated value
+
+    else
+      runner.registerError('This TableLookup is not based on two independent variables, so it is not supported with this method.')
+      false
+    end
+  end
+
+  def interpolate_from_two_ind_vars(runner, ind_var_1, ind_var_2, dep_var, input1, input2)
```
Comment (eringold) [inline resources/measures/upgrade_hvac_add_heat_pump_rtu/measure.rb:848]: add a test for this

**Review by eringold — APPROVED** [review]

**Pull request merged**